backend/firestore.rules

```diff
@@ -19,7 +19,7 @@ service cloud.firestore {
       allow read, write: if userCanAccess(resource);
 
       match /{documents=**} {
-        allow read, write: if userCanAccess(get(/databases/$(database)/documents/scenarios/$(scenario)));
+        allow read, write: if userCanAccess(getAfter(/databases/$(database)/documents/scenarios/$(scenario)));
```

Comment:
This turned out to be the key to making this work.  `getAfter` allows the rules to be run within the context of a `batch`.  Without using `getAfter`, I kept getting permissions errors because the newly duplicated scenario did not exist yet and we were trying to add facilities to it.  Note:  all other existing functionality (i.e. non-batched writes like updating cases, changing names, toggling sidebar switches, etc.) still works when using `getAfter`.